swagger/swagger-invocation/invocation-validator/src/main/java/org/apache/servicecomb/swagger/invocation/validator/ParanamerParameterNameProvider.java

```diff
@@ -16,39 +16,55 @@
  */
 package org.apache.servicecomb.swagger.invocation.validator;
 
+import java.lang.reflect.AccessibleObject;
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Method;
 import java.lang.reflect.Parameter;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
 
 import javax.validation.ParameterNameProvider;
 
 import org.apache.servicecomb.swagger.generator.core.utils.ParamUtils;
 
 
 public class ParanamerParameterNameProvider implements ParameterNameProvider {
+  private final Map<AccessibleObject, List<String>> methodCache = makeMethodCache();
+
+  protected Map<AccessibleObject, List<String>> makeMethodCache() {
+    return Collections.synchronizedMap(new WeakHashMap<AccessibleObject, List<String>>());
+  }
 
   @Override
   public List<String> getParameterNames(Constructor<?> constructor) {
-    Parameter[] parameters = constructor.getParameters();
-    List<String> parameterNames = new ArrayList<>(parameters.length);
+    List<String> parameterNames = methodCache.get(constructor);
```

Comment:
methodCache created from Collections.synchronizedMap
methodCache.get is low performance

Reply:
what can i do to implement this ?

Reply:
why not use ConcurrentHashMap/ConcurrentHashMapEx?

Reply:
done, and make ParamUtils better use